### PR TITLE
Revert "Revert "Small fix for 02340_parts_refcnt_mergetree""

### PR DESCRIPTION
**Pull request by nickitat**
Reverts ClickHouse/ClickHouse#65149

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit 6e9285791d7c8b3812f038114618bf1abf694997 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/65155/6e9285791d7c8b3812f038114618bf1abf694997/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>A Sync</td><td>If it fails, ask a maintainer for help</td><td>⏳ pending</td></tr>
<tbody>
</table>
<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>Checks correctness of the PR's body</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

**Review by hanfei1991 — APPROVED** [review]

File: tests/queries/0_stateless/02340_parts_refcnt_mergetree.sh [hunk 1]
```diff
@@ -58,7 +58,7 @@ function check_refcnt_for_table()
     $CLICKHOUSE_CLIENT -q "select table, name, refcount>=6 from system.parts where database = '$CLICKHOUSE_DATABASE' and table = '$table' and refcount >= 3"
 
     # Kill the query gracefully.
-    kill -INT $PID
```
Comment (azat) [inline tests/queries/0_stateless/02340_parts_refcnt_mergetree.sh:61]: The test works wrong if the query does not exist, instead, it's execution time should be increased, just replace `numbers(200)` with `numbers(2000)` and update the comment about max execution time to 50 seconds

And this should be enough, this is from the original failure:

```
2024.06.03 08:50:35.761910 [ 9214 ] {data_02340-uyfoemvoer} <Debug> executeQuery: (from [::1]:52074) (comment: 02340_parts_refcnt_mergetree.sh) select sleepEachRow(0.1) from data_02340 where part = 1 (stage: Complete)
...
2024.06.03 08:50:40.918544 [ 9214 ] {data_02340-uyfoemvoer} <Debug> TCPHandler: Processed in 5.176384591 sec.
...
2024.06.03 08:50:41.692615 [ 9214 ] {a74f4b5f-50cb-45ef-a42d-6d1e9b05b17d} <Debug> executeQuery: (from [::1]:52102) (comment: 02340_parts_refcnt_mergetree.sh) select table, name, refcount>=6 from system.parts where database = 'test_eyer7bd5' and table = 'data_02340' and refcount >= 3 (stage: Complete)

```

File: tests/queries/0_stateless/02340_parts_refcnt_mergetree.sh [hunk 2]
```diff
@@ -35,7 +35,7 @@ function check_refcnt_for_table()
     )
 
     # Notes:
-    # - query may sleep 0.1*(200/4)=5 seconds maximum, it is enough to check system.parts
+    # - query may sleep 0.1*(2000/4)=5 seconds maximum, it is enough to check system.parts
```
Comment (azat) [inline tests/queries/0_stateless/02340_parts_refcnt_mergetree.sh:38]:
```suggestion
    # - query may sleep 0.1*(2000/4)=50 seconds maximum, it is enough to check system.parts
```

File: tests/queries/0_stateless/02340_parts_refcnt_mergetree.sh [hunk 3]
```diff
@@ -58,7 +58,7 @@ function check_refcnt_for_table()
     $CLICKHOUSE_CLIENT -q "select table, name, refcount>=6 from system.parts where database = '$CLICKHOUSE_DATABASE' and table = '$table' and refcount >= 3"
 
     # Kill the query gracefully.
-    kill -INT $PID
+    kill -INT $PID ||:
```
Comment (nickitat) [inline tests/queries/0_stateless/02340_parts_refcnt_mergetree.sh:61]:
```suggestion
    kill -INT $PID
```

File: tests/queries/0_stateless/02340_parts_refcnt_mergetree.sh (hunk 1 above)
Comment (nickitat) [reply tests/queries/0_stateless/02340_parts_refcnt_mergetree.sh]: ok, thx, hope it is good now

**Review by azat — APPROVED** [review]

**Pull request merged**